Android/app/src/androidTest/java/tie/hackathon/travelguide/Tests/Helpers.java

```diff
@@ -0,0 +1,261 @@
+        return device.findObject(new UiSelector().resourceId(nameSpace + ":id/" + resourceId));
+    }
+
+    public static boolean isObjectWithTextDisplyed(String text) throws Exception {
+        return Helpers.checkIfUIObjectIsVisible(allOf(withText(text), isCompletelyDisplayed()), 3);
+    }
+
+    public static boolean isObjectWithTextAndIdDisplyed(String text, int rid) throws Exception {
+        return Helpers.checkIfUIObjectIsVisible(allOf(withText(text), withId(rid), isCompletelyDisplayed()), 3);
+    }
+
+    public static boolean isObjectWithIdDisplayed(int rid) throws Exception {
+        return Helpers.checkIfUIObjectIsVisible(allOf(withId(rid), isCompletelyDisplayed()), 3);
+    }
+
+    public static Matcher<View> ObjectPosition(final Matcher<View> parentMatcher, final int childPosition) {
+        return new TypeSafeMatcher<View>() {
+            @Override
+            public void describeTo(Description description) {
+                description.appendText("with " + childPosition + " child view of type parentMatcher");
+            }
+
+            @Override
+            public boolean matchesSafely(View view) {
+                if(!(view.getParent() instanceof ViewGroup)) {
+                    return parentMatcher.matches(view.getParent());
+                }
+
+                ViewGroup group = (ViewGroup) view.getParent();
+                return parentMatcher.matches(view.getParent()) && group.getChildAt(childPosition).equals(view);
+            }
+        };
+    }
+
+    public static void clickAnObjectFromTheList(int rid, int position) throws Exception {
+        onView(ObjectPosition(withId(rid), position)).perform(click());
+    }
+
+    public static boolean clickOnAView(int rid2, Matcher<View> matcher, int position) {
+        boolean found = false;
+        int i = 0;
+        int MAX_SWIPES = 2;
+        while(!found && i < MAX_SWIPES) {
+            onView(withId(rid2)).perform(swipeUp());
+            SystemClock.sleep(500);
+            try {
+                onView(matcher).check(matches(isDisplayed())).perform(actionOnItemAtPosition(position, click()));
+                found = true;
+            } catch(Exception e) {
+                // The search continues
+            }
+            i++;
+        }
+
+        if(!found) {
+            Assert.fail("The element has not been found.");
+        }
+        return found;
+    }
+
+    public static void clickOnObjectWithContentDesc(int text) throws Exception {
+        onView(withContentDescription(text)).perform(click());
+    }
+
+    public static void clickOnItemWithIdAtPosition(int rid, int position) throws Exception {
+        onView(ObjectPosition(withId(rid), position)).perform(click());
+    }
+
+    public static void typeTextOnFieldWithId(int rid, String text) throws Exception {
+        onView(withId(rid)).perform(typeText(text));
+    }
+
+    public static void tapButtonWithId(int rid) throws Exception {
```

Comment:
de ce neaparat "Button"? "onView(withId(rid))" poate fi orice, nu neaparat button. 

Daca ii puneai "onView(allIOf(withId(rid), is(instanceOf(Button.class)))", atunci da, te refereai strict la butoane.

Reply:
am sters-o. am inlocuit tapButtonWithId cu clickASpecificObjectWithId